src/plugins/legacy/polygonRoi/toolboxes/polygonRoiToolBox.cpp

```diff
@@ -135,6 +137,7 @@ polygonRoiToolBox::polygonRoiToolBox(QWidget *parent ) :
 polygonRoiToolBox::~polygonRoiToolBox()
 {
     delete viewEventFilter;
+    viewEventFilter = nullptr; // Will be tested in connect signal due to application closing
```

Comment:
It's not entirely clear how this can solve a connection-related bug. Which signal-slot connection is this referring to ?

Reply:
If something is calling `updateTableWidgetView` after the destructor has run, that is a bug that cannot be fixed this way. It is undefined behaviour to call a method on an object that has been destroyed.
(Normally Qt automatically disconnects objects that are destroyed, but this sort of bug can happen when Qt objects are deleted manually at the wrong time such as when there are pending signals in the event loop. Using `deleteLater` on Qt objects instead of `delete` avoids that).

Reply:
(I think my comment is misleading: I didn't mean to say that using `deleteLater`on `viewEventFilter`will correct the bug, because the problem is not the event filter but the toolbox itself that is being used after destruction)

Reply:
Thanks, i found a way to solve the problem. The PR is updated